src/Operations.jl

```diff
@@ -1461,8 +1516,8 @@ function pin(ctx::Context, pkgs::Vector{PackageSpec})
 
     pkgs, deps_map = _resolve(ctx.io, ctx.env, ctx.registries, pkgs, PRESERVE_TIERED, ctx.julia_version)
     update_manifest!(ctx.env, pkgs, deps_map, ctx.julia_version)
-
     new = download_source(ctx)
+    fixup_glue!(ctx.env, pkgs)
```

Comment:
Is it possible to call `fixup_glue!` from `download_source`? Or are there cases where we download, but don't want to fixup glue?

Reply:
Maybe. Or have another function that calls both of them.

Before this PR we could create the manifest without even having to download the packages. Now, we need to look up the glue packages in the downloaded packages project file since that information is not stored in the registry.

An alternative to this is to not store the glue package mapping in the manifest at all and just look it up in the package's project file in code loading. But we have so far tried to avoid that opting for storing everything required for code loading in the manifest...